src/main/java/org/openjdk/jextract/impl/StructBuilder.java

```diff
@@ -217,11 +217,15 @@ public static MemorySegment allocateArray(long len, SegmentAllocator allocator)
             """);
     }
 
-    private void emitOfAddressScoped() {
+    private void emitReinterpret() {
         appendIndentedLines("""
 
-            public static MemorySegment ofAddress(MemorySegment addr, Arena scope) {
-                return addr.reinterpret($LAYOUT().byteSize(), scope, null);
+            public static MemorySegment reinterpret(MemorySegment addr, Arena scope, Consumer<MemorySegment> cleanup) {
+                return addr.reinterpret($LAYOUT().byteSize(), scope, cleanup);
```

Comment:
Maybe crazy idea - if we did `reinterpret(Long.MAX_VALUE, ...).asSlice($LAYOUT())` that would check alignment too - not sure if that would be useful or not.

Reply:
Should the simpler reinterpret method call the array one with length of 1 ?

Reply:
> Should the simpler reinterpret method call the array one with length of 1 ?

yes, good suggestion

Reply:
As for using asSlice as an alignment check. We don't have an asSlice method in MS which takes an element count, so we would also have to create a SequenceLayout on the fly. Not sure what to do... kinda feels like we need a reinterpret overload that takes an alignment, or another more convenient way to check alignment.

I think in principle checking alignment here is a good idea, since we're 'sanitizing' a raw address here, and an alignment check is a good way to catch corrupt pointers.

Reply:
I'll leave it for now.